BrainPortal/app/helpers/userfiles_helper.rb

```diff
@@ -91,27 +92,22 @@ def data_link(file_name, userfile)
     return h(display_name) unless file_lstat.file?
 
     matched_class = SingleFile.descendants.unshift(SingleFile).find { |c| file_name =~ c.file_name_pattern }
-
-    if matched_class && matched_class <= TextFile
-      link_to h(display_name),
-              display_userfile_path(userfile,
-                :content_loader        => :collection_file,
-                :arguments             => file_name,
-                :viewer                => :text_file,
-                :viewer_userfile_class => :TextFile,
-                :content_viewer        => "off",
-              ),
-              :target => "_blank"
-    elsif matched_class && matched_class <= ImageFile
-      link_to h(display_name),
-              display_userfile_path(userfile,
-                :content_loader        => :collection_file,
-                :arguments             => file_name,
-                :viewer                => :image_file,
-                :viewer_userfile_class => :ImageFile,
-                :content_viewer        => "off",
-              ),
-              :target => "_blank"
+    viewer        = matched_class.class_viewers.first.partial rescue nil
+
+    if matched_class && viewer
+      on_click_ajax_replace(
+          { :url     => display_userfile_path(userfile,
+                                             :file_name             => file_name,
+                                             :file_size             => file_size,
+                                             :action                => :display,
+                                             :viewer                => viewer,
+                                             :viewer_userfile_class => matched_class
+                                             ),
+            :replace => "sub_viewer_filecollection_cbrain",
+          }
+        ) do
+          ("<span class=\"sub_viewable_link\">"+Pathname.new(file_name).basename.to_s+"</span>").html_safe
```

Comment:
You have the variable `display_name` already computed above that you can use in the span, no need to repeat all that basename extraction code